functions/functionsv2/slack/search.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_slack_search]
+
+// Package slack is a Cloud Function which recieves a query from
+// a Slack command and responds with the KG API result.
+package slack
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/functions-framework-go/functions"
+)
+
+const (
+	version                     = "v0"
+	slackRequestTimestampHeader = "X-Slack-Request-Timestamp"
+	slackSignatureHeader        = "X-Slack-Signature"
+)
+
+type Attachment struct {
+	Color     string `json:"color"`
+	Title     string `json:"title"`
+	TitleLink string `json:"title_link"`
+	Text      string `json:"text"`
+	ImageURL  string `json:"image_url"`
+}
+
+// Message is the a Slack message event.
+// see https://api.slack.com/docs/message-formatting
+type Message struct {
+	ResponseType string       `json:"response_type"`
+	Text         string       `json:"text"`
+	Attachments  []Attachment `json:"attachments"`
+}
+
+func init() {
+	functions.HTTP("KGSearch", kgSearch)
+	setup(context.Background())
+}
+
+// kgSearch uses the Knowledge Graph API to search for a query provided
+// by a Slack command.
+func kgSearch(w http.ResponseWriter, r *http.Request) {
+
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("failed to read body: %v", err)
+		http.Error(w, "could not read request body", 400)
+		return
+	}
+	r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+
+	if r.Method != "POST" {
+		http.Error(w, "Only POST requests are accepted", 405)
```

Comment:
nit: the http has well defined constantans for common error codes, please use those.

Reply:
Done.